src/languageSupport/languages/flux/lsp/connection.ts

```diff
@@ -188,84 +167,49 @@ class LspConnectionManager {
     })
   }
 
-  _alignInvisibleDivToEditorBlock() {
-    // elements in monaco-editor. positioned by editor.
-    const syncIcons = document.getElementsByClassName(ICON_SYNC_CLASSNAME)
-
-    // UI elements we control
-    const clickableInvisibleDiv = document.getElementById(ICON_SYNC_ID)
-    if (!syncIcons.length || !clickableInvisibleDiv) {
-      return
+  _compositionSyncStyle(startLine: number, endLine: number, synced: boolean) {
+    const classNamePrefix = synced
+      ? 'composition-sync--on'
+      : 'composition-sync--off'
+
+    // Customize the full width of Monaco editor margin using API `marginClassName`
+    // https://github.com/microsoft/monaco-editor/blob/35eb0ef/website/typedoc/monaco.d.ts#L1533
+    const startLineStyle = {
+      range: new MonacoTypes.Range(startLine, 1, startLine, 1),
+      options: {
+        marginClassName: `${classNamePrefix}--first`,
```

Comment:
`marginClassName` is a Monaco editor API that allows us to customize the left margin of the editor

Reply:
Nice find!